### PR TITLE
Improve e2e coverage for the key command

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

Improve e2e coverage for the key command:
- add a few tests for `singularity key <subcmd> --help` commands
- add tests for `singularity key newpair --but-length` commands

**This fixes or addresses the following GitHub issues:**

- Addresses #4087 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/key/key.go [hunk 1]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+	}{
+		{
+			name:          "help",
+			cmdArgs:       []string{"--help"},
+			expectedExit:  0,
+			expectedRegex: `^Upload a public key to a key server`,
+		},
+	}
+	for _, tt := range tests {
+		c.singularityResetKeyring(t) // Remove the tmp keyring before each import
```
Comment (WestleyK) [inline e2e/key/key.go]: You dont need this line here.

```suggestion
```

File: e2e/key/key.go [hunk 2]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
 
+func (c *ctx) singularityKeyPush(t *testing.T) {
+	tests := []struct {
+		name          string
+		cmdArgs       []string
+		expectedExit  int
+		expectedRegex string
+	}{
+		{
+			name:          "help",
```
Comment (WestleyK) [inline e2e/key/key.go]: Just to make it a little more clear

```suggestion
￼			name:          "push help",
```

File: e2e/key/key.go [hunk 3]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+			cmdArgs:       []string{"--help"},
+			expectedExit:  0,
+			expectedRegex: `^Download a public key from a key server`,
+		},
+	}
+	for _, tt := range tests {
+		c.singularityResetKeyring(t) // Remove the tmp keyring before each import
```
Comment (WestleyK) [inline e2e/key/key.go]: Same here.

```suggestion
```

File: e2e/key/key.go [hunk 4]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+
+func (c *ctx) singularityKeyPull(t *testing.T) {
+	tests := []struct {
+		name          string
+		cmdArgs       []string
+		expectedExit  int
+		expectedRegex string
+	}{
+		{
+			name:          "help",
```
Comment (WestleyK) [inline e2e/key/key.go]:
```suggestion
			name:          "pull help",
```

File: e2e/key/key.go [hunk 5]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+
+func (c *ctx) singularityKeyRemove(t *testing.T) {
+	tests := []struct {
+		name          string
+		cmdArgs       []string
+		expectedExit  int
+		expectedRegex string
+	}{
+		{
+			name:          "help",
```
Comment (WestleyK) [inline e2e/key/key.go]:
```suggestion
			name:          "remove help",
```

File: e2e/key/key.go [hunk 6]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+			cmdArgs:       []string{"--help"},
+			expectedExit:  0,
+			expectedRegex: `^Remove a local public key from your keyring`,
+		},
+	}
+	for _, tt := range tests {
+		c.singularityResetKeyring(t) // Remove the tmp keyring before each import
```
Comment (WestleyK) [inline e2e/key/key.go]: Same here.

```suggestion
```

File: e2e/key/key.go [hunk 7]
```diff
@@ -315,6 +316,156 @@ func (c *ctx) singularityResetKeyring(t *testing.T) {
...
+			args: []string{"newpair", "--bit-length", "0"},
+			consoleOps: []string{
+				"e2e test key",
+				"jdoe@sylabs.io",
+				" for e2e tests",
+				"e2etests",
+				"e2etests",
+				"n",
+			},
+			expectedKeyLength: 2048,
```
Comment (WestleyK) [inline e2e/key/key.go:438]: Why do you expect 2048 here?

**Review by WestleyK — CHANGES_REQUESTED** [review]
Thanks for improving this, just a couple suggestions :+1:

File: e2e/key/key.go (hunk 7 above)
Comment (gvallee) [reply e2e/key/key.go:438]: @WestleyK as far as I can tell it is the default size and what I get when running the test :)

**Review by WestleyK — APPROVED** [review]
Looks good to me! Thanks :+1:

**Pull request merged**